File: android/app/src/main/java/voidawt/image/ColorModel.java
package voidawt.image;

public abstract class ColorModel {
    protected int pixel_bits;

    public ColorModel(int bits) {
        this.pixel_bits = bits;
    }

    public abstract SampleModel createCompatibleSampleModel(int w, int h);

    public int getRGB(int pixel) {
        return pixel;
    }
}

File: src/commonMain/kotlin/awt/image/ColorModel.kt
package awt.image

expect abstract class ColorModel {
    constructor(bits: Int)

    abstract fun getRed(pixel: Int): Int
    abstract fun getGreen(pixel: Int): Int
    abstract fun getBlue(pixel: Int): Int
    abstract fun getAlpha(pixel: Int): Int

    open fun getRGB(pixel: Int): Int

    open fun createCompatibleSampleModel(w: Int, h: Int): SampleModel
}

expect val ColorModel.pixelSize: Int
expect val ColorModel.hasAlphaChannel: Boolean
expect val ColorModel.alphaPremultiplied: Boolean
expect val rgbDefaultColorModel: ColorModel